parachain/pallets/omni-account/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub use core_primitives::{Identity, MemberIdentity};
+pub use frame_system::pallet_prelude::BlockNumberFor;
+pub use pallet::*;
+
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+use sp_core::H256;
+use sp_core_hashing::blake2_256;
+use sp_std::vec::Vec;
+
+#[derive(Encode, Decode, TypeInfo, Clone, PartialEq, Eq, RuntimeDebug)]
+pub struct MemberAccount {
+	pub id: MemberIdentity,
+	pub hash: H256,
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The event type of this pallet.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// The origin which can manage the pallet.
+		type TEECallOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// The maximum number of identities an id graph can have.
+		#[pallet::constant]
+		type MaxIDGraphLength: Get<u32>;
+	}
+	pub type IDGraphMembers<T> = BoundedVec<MemberAccount, <T as Config>::MaxIDGraphLength>;
```

Comment:
`OmniAccountMembers` or maybe `SubAccounts` ?

I see no reference to omni account inside this pallet - what it really is ?

`IdGraph` -> `OmniGraph` ?
`Identity` -> `SubAccount` ?

The domain changes and it should imply vocabulary change aswell.

Do we really still link identity or maybe add identity or add (sub) account ?
What does control omni account ? Is it identity, account or maybe sub account as described in notion page ?

Reply:
You are right, we are going to have a discussion about the language/vocabulary we will use. This issue started as just a storage replacement for the current implementation (moving the storage to on-chain), but it has been evolving on the fly to adjust it with the new direction of the product and we ended up with a mix of old names (Link, Identity, etc) and new names.

It's the same for the other comment:
>Is the concept of prime identity still valid, mostly it refers to who ?
Only this and variant above mentions it.
>
>Maybe it could be changed to:
PrimeIdentityInvalid -> InvalidIdentity
PrimeIdentityNotFound -> UnknownIdGraph

We still are not 100% clear on the names we should use, but the general architecture should not change much. So we will improve the names in follow up PR's once the business defines the correct vocabulary. This will also impact the SDK